### PR TITLE
Add ShuffleNet v2

### PR DESCRIPTION
**Pull request by barrh**
Added 4 configurations: x0.5, x1, x1.5, x2
Added 2 pretrained models: x0.5, x1
Tested with Imagenet 1k.

Partially addresses: #645 

TODO: re-upload the models from my branch, and update the URLs

### CODE REVIEW

File: torchvision/models/shufflenetv2.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        self.fc = nn.Linear(self.stage_out_channels[-1], num_classes)
+
+    def forward(self, x):
+        x = self.conv1(x)
+        x = self.maxpool(x)
+        x = self.stage2(x)
+        x = self.stage3(x)
+        x = self.stage4(x)
+        x = self.conv5(x)
+        x = self.globalpool(x)
```
Comment (fmassa) [inline torchvision/models/shufflenetv2.py]: I think we should replace `globalpool` + `view` with a `x.mean([2, 3])`.

Comment (barrh) [reply torchvision/models/shufflenetv2.py]: thanks for the advice!

**Comment by barrh** [comment]
I didn't train the pretrained models. The weights were published here: https://github.com/ericsun99/Shufflenet-v2-Pytorch

**Comment by barrh** [comment]
@fmassa is there any other change request, or can we go on to merge this addition?
